### PR TITLE
fix: fixed the share button does not close when clicked outside it.

### LINKED ISSUE #3616: Bug: the share button does not close when clicked outside it
### Describe the bug

Clicking outside the share chat does not close the dialog 

https://github.com/open-sauced/app/assets/18273833/fbf610eb-288d-4c10-90a1-4a31cd6a8d74

### Steps to reproduce

1. go to star search
2. click on the share button 
3. try to click outside it, it would not close

### PR DESCRIPTION
**Pull request by AshutoshNanaware**
Fixed the issue with share chat menu not closing for outside click

## Description

This PR fixes the bug for outside click for share button. 

## Related Tickets & Documents
Fixes #3616

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/81151784/bb7f399f-7004-4fff-8b88-095c0872f2b5

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: components/StarSearch/ShareChatMenu.tsx [hunk 1]
```diff
@@ -35,8 +35,8 @@ export function ShareChatMenu({ shareUrl, copyLinkHandler, createLink }: ShareMe
   }
 
   return (
-    <DropdownMenu open={dropdownOpen} modal={false}>
-      <DropdownMenuTrigger onClick={() => setDropdownOpen(!dropdownOpen)} aria-label="Share prompt options">
+    <DropdownMenu open={dropdownOpen} modal={false} onOpenChange={(open) => setDropdownOpen(open)}>
```
Comment (a0m0rajab) [inline components/StarSearch/ShareChatMenu.tsx:38]: nit: this works the same as the next code: 
```suggestion
    <DropdownMenu open={dropdownOpen} modal={false} onOpenChange={setDropdownOpen}>
```

**Comment by a0m0rajab** [comment]
Thanks for working on the issue.

File: components/StarSearch/ShareChatMenu.tsx (hunk 1 above)
Comment (AshutoshNanaware) [reply components/StarSearch/ShareChatMenu.tsx]: @a0m0rajab done.

**Comment by jpmcb** [comment]
cc @open-sauced/engineering let's have a look at this and clean up long standing PRs 🙏🏼

**Review by nickytonline — APPROVED** [review]
Works great! Thanks for the PR @AshutoshNanaware!

🚢

**Review by jpmcb — APPROVED** [review]

**Pull request merged**